lib/media/streaming_engine.js

```diff
@@ -1262,6 +1262,17 @@ shaka.media.StreamingEngine = class {
             mediaState.stream.segmentIndex.getIteratorForTime(presentationTime);
         ref = mediaState.segmentIterator &&
             mediaState.segmentIterator.next().value;
+        // For LL-HLS we need to find the nearest INDEPENDENT segment
+        let attempt = 0;
+        const maxAttempts = 5;
+        while (ref && !ref.isIndependent() && attempt < maxAttempts) {
+          const newTime = ref.startTime - 0.1;
```

Comment:
Is looking back 0.1s 5 times the correct way to do this? 

Looking at Apple's sample manifest here:
https://developer.apple.com/documentation/http-live-streaming/enabling-low-latency-http-live-streaming-hls
They only seem to have an `INDEPENDENT` part every 1.3 seconds at best. So this might potentially miss an independent part, depending on the exact time this falls on, right?

If it's possible to have a part that has a duration that is <100ms, this approach could also be a problem the other way; that is to say, there might be the potential of "skipping over" the independent part.

Reply:
No, It's going to back 5 times, but from previous start time less 0.1 second to find the previous segment. Example:
- Segment 1 --> StartTime 0 EndTime 1, INDEPENDENT
- Segment 2 --> StartTime 1 EndTime 2
- Segment 3 --> StartTime 2 EndTime 3

If presentationTime = 3;
1 loop
2-0.1 --> Segment 2
2 loop
1-0.1 --> Segment 1 INDEPENDENT

So the loop is not 0.1 * 5ms, depends of previous segments